test/utils/wabt_engine.cpp

```diff
@@ -37,123 +43,140 @@ std::unique_ptr<WasmEngine> create_wabt_engine()
 
 bool WabtEngine::parse(bytes_view input) const
 {
-    wabt::interp::Environment env;
-
-    wabt::interp::HostModule* hostModule = env.AppendHostModule("env");
-    assert(hostModule != nullptr);
-
-    hostModule->AppendFuncExport("adler32", {{wabt::Type::I32, wabt::Type::I32}, {wabt::Type::I32}},
-        [](const wabt::interp::HostFunc*, const wabt::interp::FuncSignature*,
-            const wabt::interp::TypedValues&, wabt::interp::TypedValues&) {
-            assert(false);
-            return wabt::interp::Result::Ok;
-        });
-
-    wabt::interp::DefinedModule* module{nullptr};
+    wabt::interp::ModuleDesc module_desc;
     wabt::Errors errors;
-    const wabt::Result result = wabt::ReadBinaryInterp(
-        &env, input.data(), input.size(), wabt::ReadBinaryOptions{}, &errors, &module);
+    const wabt::Result result = wabt::interp::ReadBinaryInterp(
+        input.data(), input.size(), wabt::ReadBinaryOptions{}, &errors, &module_desc);
     return (result == wabt::Result::Ok);
 }
 
 bool WabtEngine::instantiate(bytes_view wasm_binary)
 {
-    wabt::interp::HostModule* hostModule = m_env.AppendHostModule("env");
-    assert(hostModule != nullptr);
-
-    hostModule->AppendFuncExport("adler32", {{wabt::Type::I32, wabt::Type::I32}, {wabt::Type::I32}},
-        [this](const wabt::interp::HostFunc*, const wabt::interp::FuncSignature*,
-            const wabt::interp::TypedValues& args, wabt::interp::TypedValues& results) {
-            const auto offset = args[0].value.i32;
-            const auto length = args[1].value.i32;
+    wabt::Errors errors;
+    wabt::interp::ModuleDesc module_desc;
+    const wabt::Result result = wabt::interp::ReadBinaryInterp(
+        wasm_binary.data(), wasm_binary.size(), wabt::ReadBinaryOptions{}, &errors, &module_desc);
+    if (result != wabt::Result::Ok)
+        return false;
+
+    m_module = wabt::interp::Module::New(m_store, module_desc);
+    assert(m_module);
+
+    auto host_func = wabt::interp::HostFunc::New(m_store,
```

Comment:
Is this also a smart pointer? Perhaps add an assertion here too for `host_func`.